drivers/src/trng.rs

```diff
@@ -48,6 +57,48 @@ impl Trng {
         match self {
             Self::Internal(csrng) => Ok(csrng.generate12()?.into()),
             Self::External(trng_ext) => trng_ext.generate(),
+            Self::MfgMode(rng) => Ok(rng.generate12()?.into()),
         }
     }
 }
+
+pub struct FakeRng {}
+
+impl FakeRng {
+    /// Returns a handle to the Fake RNG.
+    ///
+    /// The FakeRng will always return a constant.    
+    ///
+    /// # Errors
+    ///
+    /// None
+    pub fn new() -> CaliptraResult<Self> {
+        Ok(Self {})
+    }
+
+    pub unsafe fn assume_initialized() -> Self {
+        Self {}
+    }
+
+    /// Return 12 constant [`u32`]s.
+    ///
+    /// # Errors
+    ///
+    /// None
+    pub fn generate12(&mut self) -> CaliptraResult<[u32; 12]> {
+        self.generate()
+    }
+
+    /// Return 16 constant [`u32`]s.
+    ///
+    /// # Errors
+    ///
+    /// None
+    pub fn generate16(&mut self) -> CaliptraResult<[u32; 16]> {
+        self.generate()
+    }
+
+    fn generate<const N: usize>(&mut self) -> CaliptraResult<[u32; N]> {
+        Ok(array::from_fn(|_| 0xdeadbeef_u32))
```

Comment:
We need multiple CFI protections here check that the flag is set and that we're in debug mode.

Reply:
cfi_asserts are added to ROM main function to ensure the TRNG config correctness.